### PR TITLE
feat(onboarding): add experiment variants for progress steps

### PR DESCRIPTION
**Pull request by BohdanOne**
**Resolves**: [DT-2295](https://linear.app/prismic/issue/DT-2295/improve-progressbar-visibility)

### Description

Part of the experiment to increase Onboarding Guide discoverability is to enhance visibility of progress steps.
This PR adds experiment variant with new read-only step at the beginning, that is marked as completed.
Experiment variant has also slightly updated content.

### Checklist

- [ ] If my changes require tests, I added them.
- [ ] If my changes affect backward compatibility, it has been discussed.
- [ ] If my changes require an update to the CONTRIBUTING.md guide, I updated it.

### Preview
Zrzut ekranu 2024-08-29 o 14 15 32

### How to QA [^1]

[^1]:
	Please use these labels when submitting a review:
	:question: #ask:&ensp;Ask a question.
	:bulb: #idea:&ensp;Suggest an idea.
	:warning: #issue:&ensp;Strongly suggest a change.
	:tada: #nice:&ensp;Share a compliment.

### CODE REVIEW

File: packages/slice-machine/src/features/onboarding/content.tsx [hunk 1]
```diff
@@ -1,49 +1,103 @@
...
-  OnboardingStepType,
-  OnboardingStepContentDefinition
-> = {
-  createPageType: {
+export const onboardingExperimentSteps: OnboardingStep[] = [
+  {
+    id: "createProject",
+    title: "Create your Prismic website",
+    description: "Create and set up your project",
+    readonly: true,
```
Comment (jomifepe) [inline packages/slice-machine/src/features/onboarding/content.tsx]: 💡 `readonly` doesn't necessarily tells us that the option will be completed by default, so I would add an extra one called `defaultCompleted`/`defaultValue`/`startCompleted`.

Comment (jomifepe) [reply packages/slice-machine/src/features/onboarding/content.tsx]: 💡 Small name improvement

```suggestion
    readOnly: true,
```

File: packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx [hunk 2]
```diff
@@ -71,10 +74,14 @@ export const OnboardingProgressStepper = () => {
                   onSelect={() => showStep(step)}
                   description={step.description}
                   completed={isCompleted}
+                  readOnly={step.readonly}
                   endAdornment={
-                    <Icon name="chevronRight" size="small" color="grey11" />
+                    step.readonly !== true && (
```
Comment (dani-mp) [inline packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx]:
```suggestion
                    !step.readonly && (
```

File: packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx [hunk 3]
```diff
@@ -71,10 +74,14 @@ export const OnboardingProgressStepper = () => {
...
+                      <Icon name="chevronRight" size="small" color="grey11" />
+                    )
                   }
                 >
+                  {isOnboardingCardVisibilityExperiment && index + 1 + " "}
                   {step.title}
```
Comment (dani-mp) [inline packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx]: ⚠️ let's use a string template here to make things clearer

File: packages/slice-machine/src/features/onboarding/OnboardingProvider.tsx [hunk 4]
```diff
@@ -65,15 +73,19 @@ export const OnboardingProvider = ({
...
 
   const isStepComplete = (step: OnboardingStep) => {
     return stepStatus[step.id] ?? false;
   };
 
-  const completedStepCount = Object.values(stepStatus).filter(Boolean).length;
+  const stepsIds = steps.map(({ id }) => id);
+  const completedStepCount = Object.entries(stepStatus).filter(
+    // filtering out steps that are not part of the current experiment variant
+    ([key, value]) => stepsIds.includes(key as OnboardingStepId) && value,
```
Comment (dani-mp) [inline packages/slice-machine/src/features/onboarding/OnboardingProvider.tsx:90]: ❓ do we need the casting here?

File: packages/slice-machine/src/features/onboarding/OnboardingStepDialog/OnboardingStepDialogContent.tsx [hunk 5]
```diff
@@ -1,36 +1,34 @@
...
           <Text sx={{ marginBottom: 4 }} color="purple9" variant="bold">
-            Step {getStepIndex(step) + 1}
+            Step {getStepIndex(step.id) + 1}
           </Text>
           <Text variant="h3">{title}</Text>
-          <Content />
+          {Content && <Content />}
         </section>
-        <Video src={videoUrl} sizing="contain" autoPlay loop />
+        {typeof videoUrl == "string" && (
```
Comment (dani-mp) [inline packages/slice-machine/src/features/onboarding/OnboardingStepDialog/OnboardingStepDialogContent.tsx:28]: ❓ why checking for the string type concretely?

File: packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx (hunk 2 above)
Comment (BohdanOne) [reply packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx]: I made it like that to satisfy `strict-boolean-expressions` ts rule

File: packages/slice-machine/src/features/onboarding/OnboardingProvider.tsx (hunk 4 above)
Comment (BohdanOne) [reply packages/slice-machine/src/features/onboarding/OnboardingProvider.tsx:90]: honestly I didn't know how to do it other way, Object.entries is loosing type information inferred from zod schema :/

File: packages/slice-machine/src/features/onboarding/OnboardingStepDialog/OnboardingStepDialogContent.tsx (hunk 5 above)
Comment (BohdanOne) [reply packages/slice-machine/src/features/onboarding/OnboardingStepDialog/OnboardingStepDialogContent.tsx:28]: videoUrl can be `string | undefined`, if I go with simple `videoUrl && `.. or `!!videoUrl && ` I got TS `strict-boolean-expressions` warning

File: packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx (hunk 3 above)
Comment (BohdanOne) [reply packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx]: done

File: packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx (hunk 2 above)
Comment (dani-mp) [reply packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx]: !undefined is false, so why that rule is not happy?

File: packages/slice-machine/src/features/onboarding/OnboardingStepDialog/OnboardingStepDialogContent.tsx (hunk 5 above)
Comment (dani-mp) [reply packages/slice-machine/src/features/onboarding/OnboardingStepDialog/OnboardingStepDialogContent.tsx:28]: And I guess you can't do Boolean(videoUrl) either?

Comment (BohdanOne) [reply packages/slice-machine/src/features/onboarding/OnboardingStepDialog/OnboardingStepDialogContent.tsx:28]: this only moves the problem to another location 
Zrzut ekranu 2024-08-29 o 16 20 06

File: packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx (hunk 3 above)
Comment (dani-mp) [reply packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx]: Should we include the title as well?

Comment (BohdanOne) [reply packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx]: ok, changed again

File: packages/slice-machine/src/features/onboarding/types.ts [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import { z } from "zod";
 
 export const onboardingStepStatusesSchema = z.object({
+  createProject: z.literal(true),
```
Comment (jomifepe) [inline packages/slice-machine/src/features/onboarding/types.ts:4]: ⚠️ With the introduction of another step, can you check what happens when I have completed my onboarding with the previous structure, and then open the application with the new step? Does it keep the previous state?

File: packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx (hunk 2 above)
Comment (jomifepe) [reply packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx]: Weren't we gonna disabled this rule? 👀

Comment (dani-mp) [reply packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx]: yeah, please 🙏

File: packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx [hunk 7]
```diff
@@ -71,11 +74,16 @@ export const OnboardingProgressStepper = () => {
...
-                    <Icon name="chevronRight" size="small" color="grey11" />
+                    step.readonly !== true && (
+                      <Icon name="chevronRight" size="small" color="grey11" />
+                    )
                   }
                 >
-                  {step.title}
+                  {isOnboardingCardVisibilityExperiment
+                    ? `${index + 1} ${step.title}`
+                    : step.title}
```
Comment (dani-mp) [inline packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx:93]:
```suggestion
                  {`${isOnboardingCardVisibilityExperiment ? `${index + 1} ` : ''}${step.title}`}
```

File: packages/slice-machine/src/features/onboarding/content.tsx (hunk 1 above)
Comment (BohdanOne) [reply packages/slice-machine/src/features/onboarding/content.tsx]: agree, that doesn't tell exactly what is it used for, but I don't think it's necessary to add another, will just rename this one

File: packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx (hunk 7 above)
Comment (BohdanOne) [reply packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx:93]: Honestly, I find your suggested version much harder to read

Comment (dani-mp) [reply packages/slice-machine/src/features/onboarding/OnboardingProgressStepper.tsx:93]: that's alright, you can discard it

File: packages/slice-machine/src/features/onboarding/types.ts (hunk 6 above)
Comment (BohdanOne) [reply packages/slice-machine/src/features/onboarding/types.ts:4]: yes, I've tested that

File: packages/slice-machine/src/features/onboarding/content.tsx (hunk 1 above)
Comment (BohdanOne) [reply packages/slice-machine/src/features/onboarding/content.tsx]: changed

**Review by jomifepe — APPROVED** [review]
LGTM, great work 🚀

**Pull request merged**